### PR TITLE
thriftpool: Add package example

### PR DESCRIPTION
**Pull request by fishy**

### CODE REVIEW

File: thriftpool/doc_test.go [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+		TTLClient: thriftpool.NewTTLClient(trans, clientTTL),
+	}, nil
+}
+
+func callEndpoint(ctx context.Context, pool thriftpool.Pool) (*MyEndpointResponse, error) {
+	c, err := pool.Get()
+	if err != nil {
+		return nil, err
+	}
+	defer pool.Release(c)
```
Comment (konradreiche) [inline thriftpool/doc_test.go]: Similar question to the previous pull request, should we check errors in example code for clarification?

File: thriftpool/doc_test.go [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		maxConnections,
+		func() (thriftpool.Client, error) {
+			return newClient(remoteAddr)
+		},
+	)
+	if err != nil {
+		panic(err)
+	}
+
+	callEndpoint(context.Background(), pool)
```
Comment (konradreiche) [inline thriftpool/doc_test.go]: Same here

File: thriftpool/doc_test.go (hunk 1 above)
Comment (fishy) [reply thriftpool/doc_test.go]: Added.

File: thriftpool/doc_test.go (hunk 2 above)
Comment (fishy) [reply thriftpool/doc_test.go]: This is just an example calling the endpoint. In real code we certainly would also handle the response and errors :)

File: thriftpool/doc_test.go [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+	trans, err := thrift.NewTSocketTimeout(addr, socketTimeout)
+	if err != nil {
+		return nil, err
+	}
+	err = trans.Open()
+	if err != nil {
+		return nil, err
+	}
+	protoFactory := thrift.NewTHeaderProtocolFactory()
+	client := NewMyServiceClientFactory(trans, protoFactory)
```
Comment (pacejackson) [inline thriftpool/doc_test.go]: Would you consider using:

```suggestion
	client := NewMyServiceClient(thriftclient.NewMonitoredClientFromFactory(trans, protoFactory))
```

so we set other devs up to use that by default?

Comment (fishy) [reply thriftpool/doc_test.go]: That change was not merged in yet 😅

Comment (pacejackson) [reply thriftpool/doc_test.go]: :badpokerface:

_yet_

https://github.com/reddit/baseplate.go/commit/be09f33afa2edea73bbec51bb096648796594450

Comment (fishy) [reply thriftpool/doc_test.go]: :shakes-fist:

Comment (fishy) [reply thriftpool/doc_test.go]: And the line is soooooo long!

**Pull request merged**